File: hanoi.py
def ht(n, a, b, c) :
    if (n == 1) :
        print(str(a) + " " + str(c));
    else :
        ht(n - 1, a, c, b);
        print(str(a) + " " + str(c));
        ht(n - 1, b, a, c);
n = int(input());
power = 1;
for i in range(0, n) :
    power *= 2;
print(str(power - 1));
ht(n, 1, 2, 3);

File: hanoi.c
#include <stdio.h>
void ht(int n, int a, int b, int c) {
	if (n == 0) return ;
	else {
		ht(n - 1, a, c, b);
		printf("%d %d\n", a, c);
		ht(n - 1, b, a, c);
	}
}
int main(void) {
	int n;
	int power = 1;
	scanf_s("%d", &n);
	int i;
	printf("%d\n", (1 << i) - 1);
	ht(n, 1, 2, 3);
}
